Source/NSCollectionView.m

```diff
@@ -1114,4 +1456,558 @@ - (BOOL) wantsPeriodicDraggingUpdates
   return YES;
 }
 
+/* New methods for later versions of macOS */
+
+// 10.11 methods...
+
+/* Locating Items and Views */
+
+- (NSArray *) visibleItems
+{
+  return _visibleItems;
+}
+
+- (NSSet *) indexPathsForVisibleItems
+{
+  return _indexPathsForVisibleItems;
+}
+
+- (NSArray *) visibleSupplementaryViewsOfKind: (NSCollectionViewSupplementaryElementKind)elementKind
+{
+  return [_visibleSupplementaryViews objectForKey: elementKind];
+}
+
+- (NSSet *) indexPathsForVisibleSupplementaryElementsOfKind: (NSCollectionViewSupplementaryElementKind)elementKind
+{
+  return nil;
+}
+
+- (NSIndexPath *) indexPathForItem: (NSCollectionViewItem *)item
+{
+  return [_itemsToIndexPaths objectForKey: item];
+}
+
+- (NSIndexPath *) indexPathForItemAtPoint: (NSPoint)point
+{
+  NSIndexPath *p = nil;
+  NSEnumerator *ke = [_itemsToAttributes keyEnumerator];
+  NSCollectionViewItem *item = nil;
+  BOOL isFlipped = [self isFlipped];
+  
+  while ((item = [ke nextObject]) != nil)
+    {
+      NSCollectionViewLayoutAttributes *attr = [_itemsToAttributes objectForKey: item];
+
+      if (attr != nil)
+        {
+          NSRect f = [attr frame];
+
+          if (NSMouseInRect(point, f, isFlipped))
+            {
+              p = [self indexPathForItem: item];
+              break;
+            }
+        }
+      else
+        {
+          NSLog(@"No attributes found");
+        }
+    }
+
+  return p;
+}
+
+- (NSCollectionViewItem *) itemAtIndexPath: (NSIndexPath *)indexPath
+{
+  return [_indexPathsToItems objectForKey: indexPath];
+}
+
+- (NSView *) supplementaryViewForElementKind: (NSCollectionViewSupplementaryElementKind)elementKind 
+                                 atIndexPath: (NSIndexPath *)indexPath
+{
+  return nil;
+}
+
+- (void) scrollToItemsAtIndexPaths: (NSSet *)indexPaths 
+                    scrollPosition: (NSCollectionViewScrollPosition)scrollPosition
+{
+}
+
+/* Creating Collection view Items */
+
+- (NSCollectionViewItem *) makeItemWithIdentifier: (NSUserInterfaceItemIdentifier)identifier 
+                                     forIndexPath: (NSIndexPath *)indexPath
+{
+  NSCollectionViewItem *item =
+    [_dataSource collectionView: self itemForRepresentedObjectAtIndexPath: indexPath];
+  NSNib *nib = [self _nibForClass: [item class]];
+  BOOL loaded = [nib instantiateWithOwner: item
+                          topLevelObjects: NULL];
+
+  if (loaded == NO)
+    {
+      item = nil;
+      NSLog(@"Could not load model %@", nib);
+    }
+  
+  RELEASE(nib);
+  
+  // Add to maps...
+  [_itemsToIndexPaths setObject: indexPath
```

Comment:
You should not add these if the loading failed.

Reply:
fixed, also removed release because the nib is being AUTORELEASED per the previous change.